common/models/templates/src/common.ts

```diff
@@ -152,3 +152,21 @@ export function defaultApplyCasing(casing: CasingForm, text: string): string {
       return text.substring(0, headUnitLength).toUpperCase() .concat(text.substring(headUnitLength));
   }
 }
+
+/**
+ * An **opaque** type for a string that is exclusively used as a search key in
+ * the trie. There should be a function that converts arbitrary strings
+ * (queries) and converts them into a standard search key for a given language
+ * model.
+ *
+ * Fun fact: This opaque type has ALREADY saved my bacon and found a bug!
+ */
+export type SearchKey = string & { _: 'SearchKey'};
+
+/**
+ * A function that converts a string (word form or query) into a search key
+ * (secretly, this is also a string).
```

Comment:
```suggestion
 * (which, internally, is also a string type).
```

I realised after further review that this was just moving existing code, but feels like a good tweak, because I didn't understand why it was 'secretly'!